src/core/tests/Test_ZypperPackageManager.py

```diff
@@ -183,5 +184,51 @@ def test_install_package_failure(self):
         # test for unsuccessfully installing a package
         self.assertEquals(package_manager.install_update_and_dependencies('selinux-policy.noarch', '3.13.1-102.el7_3.16', simulate=True), Constants.FAILED)
 
+    def test_get_process_tree_from_package_manager_msg_success(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+
+        package_manager = self.container.get('package_manager')
+        self.assertIsNotNone(package_manager)
+
+        # Get this process' pid to test output with a real running process
+        test_process_pid = os.getpid()
```

Comment:
We never run unit tests on live examples. Always use mocks. Assume a pid and also the return message from the get_process-tree_cmd. See LegacyEnvLayerExtensions.py for examples, there are examples of mocking response for various commands in there

Also, make sure to run all tests within the tests folder in both /core/tests and /extension/tests with python 2.7 and any one version of python 3.x (I use python 3.8). Pycharm supports using multiple Python versions. All tests should pass (existing and new) in both versions

Reply:
Understood! I've made some changes that now use mocks based on the examples.

I've also run all the tests in both those locations with both versions and they've passed (except for the one unrelated one).